### PR TITLE
[JUJU-1459] Remove support for upstart and other service cleanup

### PR DESCRIPTION
**Pull request by wallyworld**
The primary focus is to remove support for upstart. We just support systemd now. A lot of code to detect which init system was installed can be removed.
Also, the snap service is used when installing mongo, but doesn't need to be integrated as a supported init system.
And tests were rewritten to use gomock.

The Azure and OCI provider tests and cloud init scripts were also updated to remove old series like precise and trusty etc.

## QA steps

bootstrap
juju a charm
add a manual machine
change the mongo memory profile setting
upgrade-controller

### CODE REVIEW

File: packaging/dependency/kvm.go [hunk 1]
```diff
@@ -32,16 +33,13 @@ func (dep kvmDependency) PackageList(series string) ([]packaging.Package, error)
...
 		"qemu-utils",
 		"genisoimage",
 	)
 
 	switch series {
-	case "precise", "trusty", "xenial", "bionic", "eoan":
+	case "bionic":
 		pkgList = append(pkgList, "libvirt-bin")
 	default:
 		// On focal+ virsh is provided by libvirt-clients; also we need
```
Comment (hpidcock) [inline packaging/dependency/kvm.go:45]: We should make sure we are testing jammy here but I assume the focal behaviour still works.

File: provider/oci/images_test.go [hunk 2]
```diff
@@ -8,12 +8,13 @@ import (
...
 	"github.com/golang/mock/gomock"
-	ocitesting "github.com/juju/juju/provider/oci/testing"
-	jujutesting "github.com/juju/juju/testing"
 	jc "github.com/juju/testing/checkers"
 	ociCore "github.com/oracle/oci-go-sdk/v47/core"
 	gc "gopkg.in/check.v1"
 
+	ocitesting "github.com/juju/juju/provider/oci/testing"
+	jujutesting "github.com/juju/juju/testing"
+
```
Comment (hpidcock) [inline provider/oci/images_test.go:17]: delete line

**Review by hpidcock — APPROVED** [review]

File: packaging/dependency/kvm.go (hunk 1 above)
Comment (wallyworld) [reply packaging/dependency/kvm.go:45]: yeah, and bionic is probably gone soon

**Pull request merged**